### PR TITLE
io: add lines example for `StreamReader`

### PR DESCRIPTION
**Pull request by Darksonn**
Rendered docs: [link](https://deploy-preview-5145--tokio-rs.netlify.app/tokio_util/io/struct.streamreader)

### CODE REVIEW

**Review by Noah-Kennedy — APPROVED** [review]

**Review by LucioFranco — APPROVED** [review]

File: tokio-util/src/io/stream_reader.rs [hunk 1]
```diff
@@ -250,3 +299,28 @@ where
         }
     }
 }
+
+// The code below is a manual expansion of the code that pin-project-lite would
+// generate. This is done because pin-project-lite fails by hitting the recusion
```
Comment (FrankReh) [inline tokio-util/src/io/stream_reader.rs:304]: recursion

Comment (Noah-Kennedy) [reply tokio-util/src/io/stream_reader.rs:304]: ?

Comment (FrankReh) [reply tokio-util/src/io/stream_reader.rs:304]: recusion -> recursion

Comment (Noah-Kennedy) [reply tokio-util/src/io/stream_reader.rs:304]: My mind autocorrected this lol

**Pull request merged**